core/src/main/scala/org/apache/spark/util/JsonProtocol.scala

```diff
@@ -1078,7 +1078,10 @@ private[spark] object JsonProtocol {
         val blockManagerAddress = blockManagerIdFromJson(json \ "Block Manager Address")
         val shuffleId = (json \ "Shuffle ID").extract[Int]
         val mapId = (json \ "Map ID").extract[Long]
-        val mapIndex = (json \ "Map Index").extract[Int]
+        val mapIndex = (json \ "Map Index") match {
+          case JNothing => 0
```

Comment:
nit: 0 is a legal value for the map index here. Since it's for backward compatibility, an illegal value here should make more sense. Also, it's better to comment here for emphasizing it's for backward compatibility, otherwise the fetch failed event will be dropped when SHS loading old version logs.

Reply:
Yeah that's valid concern, and caller may need to be prepared to handle the invalid value. Why not we fix it for FOLLOW-UP PR? We have been doing it even for nits. This doesn't even look to be a nit.

Reply:
Make sense. It deserves a follow-up, let me submit one soon.